src/test/csfle.rs

```diff
@@ -2839,3 +2839,120 @@ async fn bind(addr: &str) -> Result<TcpListener> {
         Ok(TcpListener::bind(addr).await?)
     }
 }
+
+// FLE 2.0 Documentation Example
+#[cfg_attr(feature = "tokio-runtime", tokio::test)]
+#[cfg_attr(feature = "async-std-runtime", async_std::test)]
+async fn fle2_example() -> Result<()> {
+    if !check_env("fle2_example", false) {
+        return Ok(());
+    }
+    let _guard = LOCK.run_exclusively().await;
+
+    // FLE 2 is not supported on Standalone topology.
+    let test_client = Client::test_builder().build().await;
+    if test_client.server_version_lt(6, 0) {
+        log_uncaptured("skipping fle2 example: server below 6.0");
+        return Ok(());
+    }
+    if test_client.is_standalone() {
+        log_uncaptured("skipping fle2 example: cannot run on standalone");
+        return Ok(());
+    }
+
+    // Drop data from prior test runs.
+    test_client
+        .database("keyvault")
+        .collection::<Document>("datakeys")
+        .drop(None)
+        .await?;
+    test_client.database("docsExamples").drop(None).await?;
+
+    // Create two data keys.
+    let ce = ClientEncryption::new(
+        test_client.clone().into_client(),
+        KV_NAMESPACE.clone(),
+        LOCAL_KMS.clone(),
+    )?;
+    let key1_id = ce.create_data_key(MasterKey::Local).run().await?;
+    let key2_id = ce.create_data_key(MasterKey::Local).run().await?;
+
+    // Create an encryptedFieldsMap.
+    let encrypted_fields_map = vec![(
```

Comment:
minor: I notice Go accepts the encrypted fields map as a map rather than a vec. I can't really tell if the spec cares which way we do this - is it specified anywhere? I don't remember if we talked about this - out of curiosity, was there a reason we took the `impl IntoIterator<Item = (String, Document)>` approach vs a map?

Reply:
I personally like the types we use for all the different configuration options. It can make it hard to reason about things when everything is just a document.

To cut back on the `to_string()` spam at least, we could accept `impl IntoIterator<Item = (impl Into<String>, Document)>` instead I believe.

Reply:
Huh, I thought I'd tried using nested `impl Foo` for something else and it wasn't a thing that worked, but it works fine here.  That removes that friction point and I has me agreeing on the types.

Reply:
... and I just realized I applied that change to #794, not this PR.  Want me to back it out and reapply here, or is it okay where it is?

Reply:
If it affects the examples here (which I think it does), then it probably makes sense to reapply here, or whichever one gets merged second. If it doesn't have any bearing on the examples, then either one works imo.

Reply:
It affects the examples in the manual much more than the example here, so I've left it in that PR.  I'll merge that one first and then update this one before merging.